apis/goharbor.io/v1alpha3/harborcluster_types.go

```diff
@@ -41,39 +41,45 @@ type Cache struct {
 	// +kubebuilder:validation:Enum=Redis
 	Kind string `json:"kind"`
 
+	// RedisSpec is the specification of redis.
 	// +kubebuilder:validation:Required
 	RedisSpec *RedisSpec `json:"redisSpec"`
 }
 
 type RedisSpec struct {
 	harbormetav1.ImageSpec `json:",inline"`
 
-	Server   *RedisServer   `json:"server,omitempty"`
+	// +kubebuilder:validation:Optional
+	// Server is the configuration of the redis server.
+	Server *RedisServer `json:"server,omitempty"`
```

Comment:
Server and Sentinel are optional?

Reply:
If the user not define these fields, the controller will use a predefined default value.